app/dashboard/static/js/app/view-socs-soc-job-kernel.2017.4.js

```diff
@@ -591,83 +230,345 @@ define([
+            .info(false)
+            .draw();
+    }
+
+    function chartCountFailed() {
+        /* The chart will not be shown */
+    }
+
+    function getBatchStatusDone(response) {
+        function parseBatchData(data) {
+            var node = document.getElementById(data.operation_id);
+            var status = (data.result[0].count == 0 ? "PASS" : "FAIL");
+            node.appendChild(ttest.statusNode(status));
+        }
+
+        response.result.forEach(parseBatchData)
+    }
+
+    function getBatchStatusFailed() {
+        console.log("getBatchStatusFailed()");
+    }
+
+    function getBatchCountDone(response) {
+        function parseBatchData(data) {
+            html.replaceContent(
+                document.getElementById(data.operation_id),
+                document.createTextNode(data.result[0].count));
+        }
+
+        response.result.forEach(parseBatchData)
+    }
+
+    function getBatchStatus(results) {
+        var batchOps;
+        var deferred;
 
-            gResultFilter.unload();
-
-            pageState['.df-success'] = {
-                type: 'attr',
-                name: 'style',
-                value: html.attrBySelector('.df-success', 'style')
-            };
-            pageState['.df-failed'] = {
-                type: 'attr',
-                name: 'style',
-                value: html.attrBySelector('.df-failed', 'style')
-            };
-            pageState['.df-unknown'] = {
-                type: 'attr',
-                name: 'style',
-                value: html.attrBySelector('.df-unknown', 'style')
-            };
-            pageState['#all-btn'] = {
-                type: 'class',
-                name: 'class',
-                value: html.attrById('all-btn', 'class')
-            };
-            pageState['#success-btn'] = {
-                type: 'class',
-                name: 'class',
-                value: html.attrById('success-btn', 'class')
-            };
-            pageState['#fail-btn'] = {
-                type: 'class',
-                name: 'class',
-                value: html.attrById('fail-btn', 'class')
-            };
-            pageState['#unknown-btn'] = {
-                type: 'class',
-                name: 'class',
-                value: html.attrById('unknown-btn', 'class')
-            };
-
-
-            Array.prototype.forEach.call(
-                document.querySelectorAll('[id^="panel-boot-"]'),
-                _saveElementState);
-
-            Array.prototype.forEach.call(
-                document.querySelectorAll('[id^="collapse-boot-"]'),
-                _saveElementState);
-
-            gSessionStorage.addObjects(pageState).save();
+        function createBatchOp(result) {
+            var job = result.job;
+            var kernel = result.kernel;
+            var branch = result.git_branch;
+            var plan = result.name;
+            var mach = gSoc;
+            var qStr;
+
+            qStr = 'job=' + job;
```

Comment:
Please use `URI.buildQuery` as in this example:
https://github.com/kernelci/kernelci-frontend/blob/master/app/dashboard/static/js/app/view-tests-job-branch-kernel-plan.2020.1.js#L362

It's a much safer way of constructing the query because it will encode special characters.  I hit an issue with the simple approach of creating a plain string when defconfig values had `+` and `=` characters in them.

Reply:
updated.